newsService/src/main/java/com/kovan/app/service/NewsService.java

```diff
@@ -38,25 +40,27 @@ public class NewsService {
     private final ObjectMapper objectMapper;
     private final NewsRepositoryService service;
     private final NewsRepository newsRepository;
+    private final NewsMapper newsMapper;
 
     private static final int DEFAULT_PAGE_SIZE = 20;
 
     LocalDate today = LocalDate.now();
     LocalDate yesterday = today.minusDays(1);
 
     public NewsService(RestTemplate restTemplate, ObjectMapper objectMapper,
-                       NewsRepositoryService service, NewsRepository newsRepository) {
+                       NewsRepositoryService service, NewsRepository newsRepository, NewsMapper newsMapper) {
         this.restTemplate = restTemplate;
         this.objectMapper = objectMapper;
         this.service = service;
         this.newsRepository = newsRepository;
+        this.newsMapper = newsMapper;
     }
     public List<NewsDto> getTopHeadlines() {
 
         List<NewsDto.Article> newsArticles;
         List<NewsDto.Article> futureArticles;
-        Optional <NewsDto> savedYesterdayNewsDto;
-        Optional <NewsDto> savedTodayNewsDto;
+        NewsDto savedYesterdayNewsDto;
```

Comment:
Use Optional instead of using blind NewsDto

Reply:
Used optional